Firestore/core/src/firebase/firestore/auth/firebase_credentials_provider_apple.mm

```diff
@@ -124,8 +124,15 @@
         }
       };
 
-  [contents_->auth getTokenForcingRefresh:contents_->force_refresh
-                             withCallback:get_token_callback];
+  // TODO(wilhuff): Need a better abstraction over a missing auth provider.
+  if (contents_->auth) {
+    [contents_->auth getTokenForcingRefresh:contents_->force_refresh
+                               withCallback:get_token_callback];
+  } else {
+    // If there's no Auth provider, call back immediately with a nil
+    // (unauthenticated) token.
+    get_token_callback(nil, nil);
```

Comment:
A couple of questions:
* is it called on the same dispatch queue as before? If not, then just to make sure, it's not going to cause any difference, right?
* auth is also accessed [here](https://github.com/firebase/firebase-ios-sdk/blob/7d30d25fcedeb277a35b471cb90068e0be3ace5b/Firestore/core/src/firebase/firestore/auth/firebase_credentials_provider_apple.mm#L39) to create a `User` object. Is that line okay?

Reply:
I think that's fine.

http://go/soverflow/2696891

IIUC, it'll return 0, which should cause FromUid to return unauthenticated.

Reply:
The first question may be a problem. Checking.

Auth calls us back on the main thread and this may fail on the async queue due to the strict no self-enqueue rule.

Reply:
OK: This is fine. I've verified in a live App that we're OK.

This is equivalent to what Core used to do:

https://github.com/firebase/firebase-ios-sdk/blob/master/Firebase/Core/FIRApp.m#L304

Reply:
Also, when calling on the stream, we specifically allow for this here:

```
  _credentials->GetToken([self](util::StatusOr<Token> result) {
    [self.workerDispatchQueue dispatchAsyncAllowingSameQueue:^{
      [self resumeStartWithToken:result];
    }];
  });
```